components/compositing/compositor.rs

```diff
@@ -396,6 +399,7 @@ impl<Window: WindowMethods> IOCompositor<Window> {
             last_composite_time: 0,
             ready_to_save_state: ReadyState::Unknown,
             scroll_in_progress: false,
+            in_scroll_transaction: None,
```

Comment:
@mbrubeck Thank you for your suggestion. I have done some additional testing to be sure, and it seems that doing anything with `in_scroll_transaction` in those functions is unnecessary. 

That is because, if those two functions are called (when the OS actually sends start/end events, which only Mac OS does at this point) the scroll transaction is, as far as I can tell, completely transparent.

To be precise, there is only one case in which the "transaction" will result in a change in the `ScrollZoomEvent` that will be sent to Webrender: When `scroll_in_progress` is false, and we receive a scroll after 80 ms. In this case a new "start event" is simulated by the transaction. 

The only risk therefore, in my opinion, is that we would send inappropriate, or double, `ScrollEventPhase::Start`. This could happen if `on_scroll_start_window_event` were to be first called, followed by the transaction code sending another start phase after that. That is currently impossible, because `on_scroll_start_window_event` will always set `scroll_in_progress` to `true`. 

An additional note: I have tried to implement the transactions with `scroll_in_progress`, however that field is used to implement a concept that overlaps, yet differs significantly from what those 'transactions' currently do. (For example, when the user has stopped actively scrolling, but Mac still sends additional scroll events that reflect the 'natural scrolling' on Mac, `scroll_in_progress` will be set to false, while we actually still need to keep the transaction open or else the scrolling will stop unnaturally on a Mac).